pyrdp/logging/log.py

```diff
@@ -61,3 +45,92 @@ def warning(*args):
 
 def error(*args):
     logging.getLogger(LOGGER_NAMES.PYRDP).error(*args)
+
+
+def convertConfig(cfg: ConfigParser):
+    """
+    Transform a config file into a dictionary.
+
+    Keys with the title format `section:subsection:subsection` are
+    transformed into nested dictionaries
+
+    # Examples
+
+    ```ini
+    [toplevel]
+    a = 1
+    b = 2
+
+    [toplevel:sublevel]
+    c = 3
+    d = 4
+    ```
+
+    would convert to
+
+    ```python
+    'toplevel': {
+        'a': 1,
+        'b': 2,
+        'sublevel': {
+            'c': 3,
+            'd': 4,
+        }
+    }
+    ````
+    """
+    def get(d: dict, key: str, create=False) -> dict:
```

Comment:
I think using a DefaultDict might simplify that code but maybe I'm wrong since it seems to be optional (`create` arg). Read about it here: https://stackoverflow.com/a/652226/262624

Reply:
It would only really remove the `if key in dict` check and add an arbitrary nesting cap, since I need to split and iterate sub dictionaries during the parsing of the ini file anyway to retrieve the right section.